File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Provider/FacebookApiInterface.java
package com.pokeapi.lpiem.pokeapiandroid.Provider;

import com.facebook.AccessToken;
import com.facebook.login.LoginResult;

public interface FacebookApiInterface {

    /**
     * Déconnexion d'un réseau social
     */
    void logout();

    /**
     * Récupération des données
     * @return l'objet du profil
     */
    Object getData(LoginResult loginResult, AccessToken accessToken);

    /**
     * Vérifie si l'utilisateur est connecté
     * @return 1=> Connecté 0=> Pas connecté
     */
    boolean isConnected();
}

File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Provider/SocialNetworks/FacebookApiInterface.kt
package com.pokeapi.lpiem.pokeapiandroid.Provider.SocialNetworks

import com.facebook.AccessToken
import com.facebook.login.LoginResult

interface FacebookApiInterface {

    /**
     * Vérifie si l'utilisateur est connecté
     * @return 1=> Connecté 0=> Pas connecté
     */
    val isConnected: Boolean

    /**
     * Déconnexion d'un réseau social
     */
    fun logout()

    /**
     * Récupération des données
     * @return l'objet du profil
     */
    fun getData(loginResult: LoginResult, accessToken: AccessToken): Any
}
